src/zeit/cms/tagging/browser/widget.py

```diff
@@ -119,31 +127,41 @@ class DisplayWidget(grokcore.component.MultiAdapter,
     grokcore.component.provides(
         zope.formlib.interfaces.IDisplayWidget)
 
+    template = zope.app.pagetemplate.ViewPageTemplateFile('display-tag.pt')
+    tag_highling_css_class = 'with-topic-page'
+
     def __init__(self, field, source, request):
         super(DisplayWidget, self).__init__(
             field,
             zope.formlib.source.IterableSourceVocabulary(source, request),
             request)
+        self.tags_with_topicpages = defaultdict(
+            str, get_tags_with_topicpages(self.context.context))
 
     def __call__(self):
-        return zope.formlib.widget.renderElement(
-            'div',
-            cssClass='keyword-widget',
-            contents=super(DisplayWidget, self).__call__(),
-            id=self.name)
-
-    def renderItems(self, value):
-        """Render items of sequence."""
-        # XXX blame formlib for having to copy this method
+        return self.template()
+
+    def _text(self, item):
+        return self.textForValue(self.vocabulary.getTerm(item))
+
+    def items(self):
         items = []
-        cssClass = self.cssClass or ''
-        if cssClass:
-            cssClass += "-item"
-        tag = self.itemTag
-        for index, item in enumerate(value):
-            term = self.vocabulary.getTerm(item)
-            items.append(zope.formlib.widget.renderElement(
-                tag,
-                cssClass=cssClass,
-                contents=xml.sax.saxutils.escape(self.textForValue(term))))
+        Tag = namedtuple('Tag', ['text', 'link', 'css_class'])
+        for item in self._getFormValue():
+            text = self._text(item)
+            link = self.tags_with_topicpages[item.uniqueId]
+            css_class = self.tag_highling_css_class if link else ''
+            items.append(Tag(text, link, css_class))
         return items
+
+
+def get_tags_with_topicpages(article):
```

Comment:
Diese Funktion sollte man denke ich auf den `ITagger` mit draufmachen. Dann kann die echte Implementierung nach zeit.retresco wandern, und es für die Tests zu mocken müsste sich auch vereinfachen.

Evtl. könnte man es sowas wie `ITagger.links` nennen ("dict from tag codes to topicpage URL, or None"), das ist kürzer als "things with topicpages". :) Vmtl. sinnvollerweise mit `@cachedproperty`. Und dann kann auch das `defaultdict` ausm DisplayWidget rausfliegen, wahrscheinlich.